src/parser/expression.js

```diff
@@ -390,8 +390,8 @@ pp.parseSubscripts = function (base, startPos, startLoc, noCalls) {
     } else if (this.hasPlugin("lightscript") && !noCalls && this.eat(tt.tilde)) {
       const node = this.startNodeAt(startPos, startLoc);
       node.left = base;
-      // allow Identifier or MemberExpression, but not calls
```

Comment:
Restored, but above the newly added line and also calling out that `this` is allowed.